Libraries/CommonFunctions.psm1

```diff
@@ -1956,61 +1956,72 @@ function Enable-RootUser {
     return $deploymentResult
 }
 
-function IsGreaterKernelVersion() {
+function Is-DpdkCompatible() {
     param (
-        [string] $actualKernelVersion,
-        [string] $detectedDistro
+        [string] $KernelVersion,
+        [string] $DetectedDistro,
+        [string] $CompatibleDistro
     )
 
-    # Supported Distro and kernel version for DPDK on Azure
-    # https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk
-    $SUPPORTED_DISTRO_KERNEL = @{
-        "UBUNTU" = "4.15.0-1015-azure";
-        "SLES" = "4.12.14-5.5-azure";
-        "SLES 15" = "4.12.14-5.5-azure";
-        "SUSE" = "4.12.14-5.5-azure";
-        "REDHAT" = "3.10.0-862.9.1.el7";
-        "CENTOS" = "3.10.0-862.3.3.el7";
-    }
-
-    if ($SUPPORTED_DISTRO_KERNEL.Keys -contains $detectedDistro) {
-        $supportKernelVersions = $SUPPORTED_DISTRO_KERNEL[$detectedDistro] -split "[\.\-]+"
-        $actualKernelVersions = $actualKernelVersion -split "[\.\-]+"
-        for($i=0; $i -lt $supportKernelVersions.Length;$i++) {
-            try {
-                    $supportKernelVersions[$i] = [int]$supportKernelVersions[$i]
-                } catch {
-                    $supportKernelVersions[$i] = 0
-                    continue
-                }
+    if ($CompatibleDistro) {
+        if ($CompatibleDistro.Contains($DetectedDistro)) {
+            Write-LogInfo "Confirmed supported distro: $DetectedDistro"
+            return $true
+        } else {
+            Write-LogWarn "Unsupported distro: $DetectedDistro"
+            return $false
         }
-        for($i=0; $i -lt $actualKernelVersions.Length;$i++) {
-            try {
-                    $actualKernelVersions[$i] = [int]$actualKernelVersions[$i]
-                } catch {
-                    $actualKernelVersions[$i] = 0
-                    continue
-                }
+    } else {
+        # Supported Distro and kernel version for DPDK on Azure
+        # https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk
+        $SUPPORTED_DISTRO_KERNEL = @{
+            "UBUNTU" = "4.15.0-1015-azure";
+            "SLES" = "4.12.14-5.5-azure";
+            "SLES 15" = "4.12.14-5.5-azure";
+            "SUSE" = "4.12.14-5.5-azure";
+            "REDHAT" = "3.10.0-862.9.1.el7";
+            "CENTOS" = "3.10.0-862.3.3.el7";
         }
 
-        $array_count = $actualKernelVersions.Length
-        if ($supportKernelVersions.Length -gt $actualKernelVersions.Length) {
-            $array_count = $supportKernelVersions.Length
-        }
+        if ($SUPPORTED_DISTRO_KERNEL.Keys -contains $DetectedDistro) {
+            $supportKernelVersions = $SUPPORTED_DISTRO_KERNEL[$DetectedDistro] -split "[\.\-]+"
+            $KernelVersions = $KernelVersion -split "[\.\-]+"
+            for($i=0; $i -lt $supportKernelVersions.Length;$i++) {
+                try {
+                        $supportKernelVersions[$i] = [int]$supportKernelVersions[$i]
```

Comment:
Please check indentation

Reply:
double check, no issue here, since those code moved to else block, thanks!